src/core/unit.c

```diff
@@ -4831,3 +4831,72 @@ int unit_fork_helper_process(Unit *u, pid_t *ret) {
         *ret = pid;
         return 1;
 }
+
+static void unit_update_dependency_mask(Unit *u, UnitDependency d, Unit *other, UnitDependencyInfo di) {
+        assert(u);
+        assert(d >= 0);
+        assert(d < _UNIT_DEPENDENCY_MAX);
+        assert(other);
+
+        if (di.origin_mask == 0 || di.destination_mask == 0) {
+                /* No bit set anymore, let's drop the whole entry */
+                assert_se(hashmap_remove(u->dependencies[d], other));
+                log_unit_error(u, "Removing %s dependency %s vs. %s", unit_dependency_to_string(d), u->id, other->id);
```

Comment:
`log_unit_debug`? Also this is pretty unclear. Maybe `("%s lost %s=%s dependency",  unit_dependency_to_string(d), u->id, other->id)`? (I think it's better to keep the source unit first in the message for easier parsing, hence this convoluted sentence. Maybe you can come up with something better.)

Reply:
uh, this is debugging left-over...

Reply:
but we might as well keep it, and fix the log message as you say.